lib/puppet/pops/issue_reporter.rb

```diff
@@ -106,11 +102,30 @@ def self.create_exception(exception_class, emit_message, formatter, diagnostic)
     file = diagnostic.file
     file = (file.is_a?(String) && file.empty?) ? nil : file
     line = pos = nil
+
     if diagnostic.source_pos
       line = diagnostic.source_pos.line
       pos = diagnostic.source_pos.pos
     end
-    exception_class.new(format_with_prefix(emit_message, formatter.format_message(diagnostic)), file, line, pos, nil, diagnostic.issue.issue_code, diagnostic.arguments)
+
+    exception = exception_class.new(
+                  format_with_prefix(emit_message, formatter.format_message(diagnostic)),
+                  file,
+                  line,
+                  pos,
+                  nil,
+                  diagnostic.issue.issue_code,
+                  diagnostic.arguments)
+
+    if diagnostic.exception
+      exception.set_backtrace(diagnostic.exception.backtrace)
+    end
+
+    if diagnostic.exception && diagnostic.exception.respond_to?(:puppet_stacktrace)
+      exception.puppet_stacktrace = diagnostic.exception.puppet_stacktrace
```

Comment:
just before I pushed this up I noticed this had a typo, so it must not be necessary for the case I was testing.

need to figure out a way to verify that we don't have to do shenanigans like this at various other points in our error handling code - we like to repackage exceptions.